presto-common/src/main/java/com/facebook/presto/common/type/TypeSignatureBase.java

```diff
@@ -39,33 +43,58 @@ public static TypeSignatureBase of(QualifiedObjectName name)
         return new TypeSignatureBase(name);
     }
 
+    public static TypeSignatureBase of(UserDefinedType userDefinedType)
+    {
+        return new TypeSignatureBase(userDefinedType.getUserDefinedTypeName(), userDefinedType.getPhysicalTypeSignature().getTypeSignatureBase().getStandardTypeBase());
+    }
+
     private TypeSignatureBase(String standardTypeBase)
     {
         checkArgument(standardTypeBase != null, "standardTypeBase is null");
         checkArgument(!standardTypeBase.isEmpty(), "standardTypeBase is empty");
         checkArgument(standardTypeBase.chars().noneMatch(c -> c == '.'), "Standard type %s should not have '.' in it", standardTypeBase);
         checkArgument(validateName(standardTypeBase), "Bad characters in base type: %s", standardTypeBase);
         this.standardTypeBase = Optional.of(standardTypeBase);
-        this.typeBase = Optional.empty();
+        this.typeName = Optional.empty();
     }
 
-    private TypeSignatureBase(QualifiedObjectName typeBase)
+    private TypeSignatureBase(QualifiedObjectName typeName)
     {
-        checkArgument(typeBase != null, "typeBase is null");
-        checkArgument(validateName(typeBase.getObjectName()), "Bad characters in base type: %s", typeBase);
+        checkArgument(typeName != null, "typeName is null");
+        checkArgument(validateName(typeName.getObjectName()), "Bad characters in typeName: %s", typeName);
         this.standardTypeBase = Optional.empty();
-        this.typeBase = Optional.of(typeBase);
+        this.typeName = Optional.of(typeName);
+    }
+
+    private TypeSignatureBase(QualifiedObjectName typeName, String standardTypeBase)
+    {
+        checkArgument(typeName != null && standardTypeBase != null, "typeName or standardTypeBase is null");
```

Comment:
It can only be a standard type?

Reply:
The "standard type" here refers to all types defined in `StandardTypes.java` which includes all presto types.